File: src/19장/19-4.py
s=str(input())
bomb=str(input())
left=[]

start=0;end=len(s)-1
while(start<=end):
  tof=True
  left.append(s[start]);start+=1
  if(len(left)>=len(bomb)):
    for i in range(len(bomb)):
      if bomb[i]!= left[len(left)-len(bomb)+i]:
        tof=False
        break
    if tof==True:
      for i in range(len(bomb)):
        left.pop()

if len(left)==0 : print("FRULA")
else :
  for i in range(len(left)):
    print(left[i],end='')

File: src/19장/19-4.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		String s = br.readLine();
		String bomb = br.readLine();
		Stack<Character> left = new Stack<Character>();
		for (int i = 0; i < s.length(); i++) {
			left.push(s.charAt(i));
			if (left.size() >= bomb.length()) {
				boolean tof = true;
				for (int j = 0; j < bomb.length(); j++) {
					if (left.get(left.size() - bomb.length() + j) != bomb.charAt(j)) {
						tof = false;
						break;                                            
					}
				}
				if (tof) {
					for (int j = 0; j < bomb.length(); j++) {
						left.pop();
					}
				}
			}
		}
		StringBuilder sb = new StringBuilder();
		for(char ch : left) {
			sb.append(ch);
		}
		System.out.println(sb.length() > 0 ? sb.toString() : "FRULA");
	}
}